bleak/backends/bluezdbus/manager.py

```diff
@@ -946,36 +951,38 @@ def _parse_msg(self, message: Message):
 
                 if interface == defs.DEVICE_INTERFACE:
                     # handle advertisement watchers
+                    device_path = message_path
 
                     self._run_advertisement_callbacks(
-                        message.path, cast(Device1, self_interface), changed.keys()
+                        device_path, cast(Device1, self_interface), changed.keys()
                     )
 
                     # handle device condition watchers
-                    for condition_callback in self._condition_callbacks.get(
-                        message.path, ()
-                    ):
-                        condition_callback()
+                    condition_callbacks = self._condition_callbacks.get(device_path)
+                    if condition_callbacks:
+                        for condition_callback in condition_callbacks:
+                            condition_callback(unpacked_changed)
 
                     # handle device connection change watchers
-
                     if "Connected" in changed:
-                        for (
-                            device_path,
-                            on_connected_changed,
-                            _,
-                        ) in self._device_watchers.copy():
-                            # callbacks may remove the watcher, hence the copy() above
-                            if message.path == device_path:
-                                on_connected_changed(self_interface["Connected"])
+                        watchers = self._device_watchers.get(device_path)
+                        if watchers:
+                            # callbacks may remove the watcher, hence the view
+                            for watcher in watchers[:]:
+                                watcher.on_connected_changed(
+                                    self_interface["Connected"]
+                                )
```

Comment:
This looks leftover from the previous PR and should not be needed.

Reply:
Just realized this is still needed put back in 9874de6cdd8f52656d6e4b1ef129cdb64f8484ee

Reply:
My mistake, you are right.